### PR TITLE
Add Edit command

### PR DESCRIPTION
**Pull request by wapisai**
Added income and family fields to edit. #69 
Yet to add additional notes and buyer/seller tags.

Added FamilyTest and IncomeTest #100

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 1]
```diff
@@ -76,17 +80,16 @@ public void parse_invalidPreamble_failure() {
 
         // invalid arguments being parsed as preamble
         assertParseFailure(parser, "1 some random string", MESSAGE_INVALID_FORMAT);
-
-        // invalid prefix being parsed as preamble
-        assertParseFailure(parser, "1 i/ string", MESSAGE_INVALID_FORMAT);
```
Comment (UdhayaShan1) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:81]: 👍

File: src/test/java/seedu/address/model/person/IncomeTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    @Test
+    public void equals() {
+        Income income = new Income("999");
+
+        // same values -> returns true
+        assertTrue(income.equals(new Income("999")));
+
+        // same object -> returns true
+        assertTrue(income.equals(income));
+
```
Comment (UdhayaShan1) [inline src/test/java/seedu/address/model/person/IncomeTest.java:69]: thanks for adding

File: src/test/java/seedu/address/model/person/FamilyTest.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        assertThrows(NumberFormatException.class, () -> new Family(invalidFamily));
+    }
+
+    @Test
+    public void isValidFamily() {
+        // null family size
+        assertThrows(NumberFormatException.class, () -> Family.isValidFamily(null));
+
+        // invalid family sizes
+        assertFalse(Family.isValidFamily("0")); // less than 1
```
Comment (UdhayaShan1) [inline src/test/java/seedu/address/model/person/FamilyTest.java:52]: thanks for adding

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 4]
```diff
@@ -259,8 +265,10 @@ public String toString() {
             return new ToStringBuilder(this)
                     .add("name", name)
                     .add("phone", phone)
+                    .add("income", income)
                     .add("email", email)
```
Comment (UdhayaShan1) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:269]: 👍

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 5]
```diff
@@ -52,12 +56,18 @@ public EditCommand parse(String args) throws ParseException {
...
         }
+        if (argMultimap.getValue(PREFIX_INCOME).isPresent()) {
+            editPersonDescriptor.setIncome(ParserUtil.parseIncome(argMultimap.getValue(PREFIX_INCOME).get()));
+        }
         if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
             editPersonDescriptor.setEmail(ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get()));
         }
         if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
             editPersonDescriptor.setAddress(ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get()));
         }
```
Comment (UdhayaShan1) [inline src/main/java/seedu/address/logic/parser/EditCommandParser.java:67]: 👍

**Review by UdhayaShan1 — APPROVED** [review]
LGTM, but pls do on another branch instead of master

**Review by 4llysa — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/logic/commands/EditCommand.java
Comment (jyztintan) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:1]: Thanks for adding the new attributes into the edit command!

File: src/main/java/seedu/address/model/person/Income.java [hunk 6]
```diff
@@ -21,6 +22,7 @@ public class Income {
      */
     public Income(String incomeValue) {
         requireNonNull(incomeValue);
+        checkArgument(isValidIncome(incomeValue), MESSAGE_CONSTRAINTS);
```
Comment (jyztintan) [inline src/main/java/seedu/address/model/person/Income.java:25]: Good spot, thanks for adding the check 👍

File: src/test/java/seedu/address/model/person/FamilyTest.java
Comment (jyztintan) [inline src/test/java/seedu/address/model/person/FamilyTest.java:1]: Thanks for updating the unit tests for Family!

File: src/test/java/seedu/address/model/person/FamilyTest.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    public void constructor_invalidFamily_throwsNumberFormatException2() {
+        String invalidFamily = " ";
+        assertThrows(NumberFormatException.class, () -> new Family(invalidFamily));
+    }
+
+    @Test
+    public void constructor_invalidFamily_throwsNumberFormatException3() {
+        String invalidFamily = " family"; //alphabets
+        assertThrows(NumberFormatException.class, () -> new Family(invalidFamily));
+    }
```
Comment (jyztintan) [inline src/test/java/seedu/address/model/person/FamilyTest.java:32]: Nitpick: maybe can standardise the test comments to a new line 
```suggestion
    @Test
    public void constructor_invalidFamily_throwsNumberFormatException3() {
        //invalid input containing alphabets
        String invalidFamily = " family";
        assertThrows(NumberFormatException.class, () -> new Family(invalidFamily));
    }
```

**Review by jyztintan — APPROVED** [review]
LGTM THANKS! 😎

File: src/test/java/seedu/address/model/person/FamilyTest.java (hunk 7 above)
Comment (UdhayaShan1) [reply src/test/java/seedu/address/model/person/FamilyTest.java]: Good spot! Committed

**Comment by UdhayaShan1** [comment]
LGTM will merge

**Pull request merged**